From 8c8ae62f56724a3968b46ed8c6e1a6359fae9d81 Mon Sep 17 00:00:00 2001
Message-Id: <8c8ae62f56724a3968b46ed8c6e1a6359fae9d81.1727076249.git.mykyta_poturai@epam.com>
From: Mykyta Poturai <mykyta_poturai@epam.com>
Date: Mon, 23 Sep 2024 10:23:53 +0300
Subject: [PATCH] [PATCH 2/5] PCIe/MSI support

ITS part is left untouched as ITS is handled by Xen.

Signed-off-by: Yoshihiro Shimoda <yoshihiro.shimoda.uh@renesas.com>
Signed-off-by: Oleksandr Andrushchenko <oleksandr_andrushchenko@epam.com>
Signed-off-by: Mykyta Poturai <mykyta_poturai@epam.com>
---
 drivers/pci/controller/dwc/pcie-rcar-gen4.c |  3 +++
 drivers/pci/controller/dwc/pcie-rcar-gen4.h | 11 +++++++++++
 2 files changed, 14 insertions(+)

diff --git a/drivers/pci/controller/dwc/pcie-rcar-gen4.c b/drivers/pci/controller/dwc/pcie-rcar-gen4.c
index 0cadcb1d51d2..8adaff9bff09 100644
--- a/drivers/pci/controller/dwc/pcie-rcar-gen4.c
+++ b/drivers/pci/controller/dwc/pcie-rcar-gen4.c
@@ -626,6 +626,9 @@ void rcar_gen4_pcie_initial(struct rcar_gen4_pcie *rcar, bool rc)
 		val |= CLK_REG | CLK_PM;
 		writel(val, rcar->base + PCIEPWRMNGCTRL);
 
+		writel(AXIINTCADDR_VAL, rcar->base + AXIINTCADDR);
+		writel(AXIINTCCONT_VAL, rcar->base + AXIINTCCONT);
+
 		/* MSI Enable */
 		val = dw_pcie_readl_dbi(dw, MSICAP0F0);
 		val |= MSIE;
diff --git a/drivers/pci/controller/dwc/pcie-rcar-gen4.h b/drivers/pci/controller/dwc/pcie-rcar-gen4.h
index 8056b28c4a9d..4477b33bf931 100644
--- a/drivers/pci/controller/dwc/pcie-rcar-gen4.h
+++ b/drivers/pci/controller/dwc/pcie-rcar-gen4.h
@@ -14,6 +14,17 @@
 
 #include "pcie-designware.h"
 
+#define AXIINTCADDR		0x0a00	/* 0x6c00 */
+#define AXIINTCADDR_VAL		0xf1050040	/* FIXME */
+#define AXIINTCCONT		0x0a04	/* 0x6c04 */
+#define AXIINTCCONT_VAL		(BIT(31) | GENMASK(11, 2))
+
+/* PCIEC PHY */
+#define RCVRCTRLP0		0x0040
+#define  PHY0_RX1_TERM_ACDC	BIT(14)
+#define  PHY0_RX0_TERM_ACDC	BIT(13)
+
+#define REFCLKCTRLP0		0x0B8
 /* PCI Express capability */
 #define EXPCAP(x)		(0x0070 + (x))
 /* ASPM L1 PM Substates */
-- 
2.34.1

